be/src/exec/spill/input_stream.cpp

```diff
@@ -121,26 +123,34 @@ StatusOr<ChunkUniquePtr> UnionAllSpilledInputStream::get_next(workgroup::YieldCo
 // The raw chunk input stream. all chunks are in memory.
 class RawChunkInputStream final : public SpillInputStream {
 public:
-    RawChunkInputStream(std::vector<ChunkPtr> chunks, Spiller* spiller) : _chunks(std::move(chunks)) {}
+    RawChunkInputStream(std::vector<ChunkPtr> chunks, Spiller* spiller) : _chunks(std::move(chunks)) {
+        for (const auto& chunk : _chunks) {
+            _total_rows += chunk->num_rows();
+        }
+    }
     StatusOr<ChunkUniquePtr> get_next(workgroup::YieldContext& yield_ctx, SerdeContext& ctx) override;
 
     bool is_ready() override { return true; };
     void close() override{};
 
 private:
-    size_t read_idx{};
+    size_t _total_rows{};
+    size_t _read_rows{};
+    size_t _read_idx{};
     std::vector<ChunkPtr> _chunks;
     DECLARE_RACE_DETECTOR(detect_get_next)
 };
 
 StatusOr<ChunkUniquePtr> RawChunkInputStream::get_next(workgroup::YieldContext& yield_ctx, SerdeContext& context) {
     RACE_DETECT(detect_get_next);
-    if (read_idx >= _chunks.size()) {
+    if (_read_idx >= _chunks.size()) {
+        DCHECK_EQ(_total_rows, _read_rows);
         return Status::EndOfFile("eos");
     }
     // TODO: make ChunkPtr could convert to ChunkUniquePtr to avoid unused memory copy
-    auto res = std::move(_chunks[read_idx++])->clone_unique();
-    _chunks[read_idx - 1].reset();
+    auto res = std::move(_chunks[_read_idx++])->clone_unique();
+    _read_rows += res->num_rows();
+    _chunks[_read_idx - 1].reset();
```

Comment:
why clone the chunk instead of moving the chunk directly？

Reply:
_chunks[i] is std::shared_ptr<Chunk> but Merger require a std::unique_ptr<Chunk>

Reply:
In a broadcast join, chunks that are not flushed to disk will be in memory. A certain join prober needs to clone this chunk and cannot modify them directly. This is because other join prober will also access them.